test/src/test/resources/fixtures/basic/service-types-are-filtered.yml

```diff
@@ -0,0 +1,174 @@
+      id: ID
+    }
+    union TestResult = Test | QueryError
+  lab: |-
+    type LabQueryErrorExtension implements QueryErrorExtension {
+      statusCode: Int
+      errorType: String
+    }
+  shared: |-
+    interface QueryErrorExtension {
+      """A numerical code (such as a HTTP status code) representing the error category"""
+      statusCode: Int
+      """Application specific error type"""
+      errorType: String
+    }
+    type GenericQueryErrorExtension implements QueryErrorExtension {
+      statusCode: Int
+      errorType: String
+    }
+    type QueryError {
+      """The ID of the object that would have otherwise been returned if not for the query error"""
+      identifier: ID
+      "A message describing the error"
+      message: String
+      "Use this to put extra data on the error if required"
+      extensions: [QueryErrorExtension!]
+    }
+underlyingSchema:
+  test: |-
+    type Query {
+      test: TestResult
+    }
+    type Test {
+      id: ID
+    }
+    union TestResult = Test | QueryError
+    interface QueryErrorExtension {
+      """A numerical code (such as a HTTP status code) representing the error category"""
+      statusCode: Int
+      """Application specific error type"""
+      errorType: String
+    }
+    type GenericQueryErrorExtension implements QueryErrorExtension {
+      statusCode: Int
+      errorType: String
+    }
+    type QueryError {
+      """The ID of the object that would have otherwise been returned if not for the query error"""
+      identifier: ID
+      "A message describing the error"
+      message: String
+      "Use this to put extra data on the error if required"
+      extensions: [QueryErrorExtension!]
+    }
+  lab: |-
+    type Query {
+      echo: String
+    }
+    interface QueryErrorExtension {
+      """A numerical code (such as a HTTP status code) representing the error category"""
+      statusCode: Int
+      """Application specific error type"""
+      errorType: String
+    }
+    type LabQueryErrorExtension implements QueryErrorExtension {
+      statusCode: Int
+      errorType: String
+    }
+    type QueryError {
+      """The ID of the object that would have otherwise been returned if not for the query error"""
+      identifier: ID
+      "A message describing the error"
+      message: String
+      "Use this to put extra data on the error if required"
+      extensions: [QueryErrorExtension!]
+    }
+  shared: |-
+    type Query {
+      echo: String
+    }
+query: |
+  query {
+    test {
+      ... on Test {
+        id
+      }
+      ... on QueryError {
+        extensions {
+          ...GenericQueryErrorExtensions
+        }
+      }
+    }
+  }
+
+  fragment GenericQueryErrorExtensions on QueryErrorExtension {
+    errorType
+    ... on LabQueryErrorExtension {
+      statusCode
+    }
+  }
+variables: {}
+serviceCalls:
+  current:
+    - serviceName: test
+      request:
+        query: |
+          query nadel_2_service {
+            foo
```

Comment:
where is `foo` in the schema?

Reply:
Oh yea good point, these tests are actually disabled for the current gen engine. So this section is actually obsolete. I'll delete it.